src/CMakeLists.txt

```diff
@@ -512,8 +512,9 @@ set_target_properties(ceph-osd PROPERTIES
   POSITION_INDEPENDENT_CODE ${EXE_LINKER_USE_PIE})
 install(TARGETS ceph-osd DESTINATION bin)
 
-if (WITH_CEPHFS)
-  add_subdirectory(mds)
+add_subdirectory(mds)
+
+if(WITH_MDS)
```

Comment:
i think add_subdirectory(mds) should still be inside WITH_MDS, so that if something else depends on the `mds` target it fails with a cmake error instead of building the mds anyway

Reply:
@cbodley 
This will only work if we also use @tchaikov 's PR #25255 which will allow building `ceph-dencoder`
without `libmds`